nni/compression/pytorch/compressor.py

```diff
@@ -608,24 +626,34 @@ def quant_backward(tensor, grad_output, quant_type):
         tensor
             gradient of the input of quantization operation
         """
+        tensor_q = QuantGrad._quantize(tensor, scale, zero_point)
+        mask = (tensor_q < qmin) | (tensor_q > qmax)
+        grad_output[mask] = 0
         return grad_output
 
     @staticmethod
     def forward(ctx, tensor, quant_type, wrapper, **kwargs):
-        ctx.save_for_backward(tensor, torch.Tensor([quant_type]))
         if quant_type == QuantType.QUANT_INPUT:
```

Comment:
Suggest changing defination of QuantType to 'input', 'weight' and 'output'.

Reply:
Um.., I dont think it is relavant with this PR. May be you need to update all definition for quant type in another pr

Reply:
could you help update line 583-585 to update the values from 0, 1, 2 to 'input', 'weight', 'output'? we have checked the code, it is safe to make this modification :)

Reply:
Ok～ 👌